lib/puppet/provider/vcsrepo.rb

```diff
@@ -16,18 +16,26 @@ def check_force
 
   private
 
-  def set_ownership
+  def set_ownership_and_permissions
     owner = @resource.value(:owner) || nil
     group = @resource.value(:group) || nil
+    mode = @resource.value(:mode) || nil
     excludes = @resource.value(:excludes) || nil
+    # Change the permission on the repo itself.
+    # The VCS should maintain permissions on files within the checkout.
+    FileUtils.chmod(mode, @resource.value(:path)) unless mode.nil?
+    # We might have no work to do, and this makes it easier for the callers.
+    return if owner.nil? && group.nil?
+
     if excludes.nil? || excludes.empty?
-      FileUtils.chown_R(owner, group, @resource.value(:path))
+      FileUtils.chown_R(owner, group, @resource.value(:path)) unless owner.nil? && group.nil?
```

Comment:
This change is useless to avoid noop chown with the return you added on line 28.

Reply:
Fixed in rebase